### PR TITLE
E06 u03 t01/course enrollment

### PR DESCRIPTION
**Pull request by lazasa**
- Unroll and enroll course
- Course interface

### CODE REVIEW

File: src/app/services/course-enrollment.service.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    // Use token
+    return this.#http.put(`${environment.API_URL}/api/course/enroll/${courseId}`, {})
+  }
+
+  unrollCourse (courseId: string) {
+    // Use token
+    return this.#http.delete(`${environment.API_URL}/api/course/dropp-out/${courseId}`)
+  }
+
+  constructor() { }
```
Comment (martinalbrnz) [inline src/app/services/course-enrollment.service.ts:23]: U can omit the constructor if u r not using it

File: src/app/services/course-enrollment.service.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  #http = inject(HttpClient)
+  // #token = inject(TokenService)
+
+  enrollCourse (courseId: string) {
+    // Use token
+    return this.#http.put(`${environment.API_URL}/api/course/enroll/${courseId}`, {})
+  }
+
+  unrollCourse (courseId: string) {
+    // Use token
```
Comment (martinalbrnz) [inline src/app/services/course-enrollment.service.ts:19]: Make an interceptor to inject the token, so u don't have to manually inject it on every request

**Review by martinalbrnz — APPROVED** [review]
Left some obs, but it's ok to merge

File: src/app/services/course-enrollment.service.ts (hunk 2 above)
Comment (lazasa) [reply src/app/services/course-enrollment.service.ts:19]: Ok, I'll be fixing this and then I merge

**Pull request merged**